crates/language/src/language_settings.rs

```diff
@@ -63,7 +63,13 @@ pub struct AllLanguageSettings {
     pub edit_predictions: EditPredictionSettings,
     pub defaults: LanguageSettings,
     languages: HashMap<LanguageName, LanguageSettings>,
-    pub(crate) file_types: FxHashMap<Arc<str>, GlobSet>,
+    pub(crate) file_types: FxHashMap<Arc<str>, LanguageCustomFileTypes>,
```

Comment:
Let's put this back to just a `GlobSet` - as mentioned above, we don't need to store the patterns themselves - a glob pattern is not something that we want to compare the length of. The length to compare is the length of the actual path suffix that matched a pattern.

Reply:
Totally fair. This is what derailed me last year when I first attempted this: The change to add glob support was added after my initial attempt and I couldn't get my head around how to handle that change. I've reverted this change, so this PR no longer touches this file at all.